kwola/tasks/RunTestingStep.py

```diff
@@ -279,19 +280,35 @@ def runTestingStep(configDir, testingStepId, shouldBeRandom=False, generateDebug
             bug.stepNumber = stepNumber
             bug.error = error
             bug.testingRunId = testStep.testingRunId
-            bug.saveToDisk(config, overrideSaveFormat="json", overrideCompression=0)
-            bug.saveToDisk(config)
 
-            bugTextFile = os.path.join(config.getKwolaUserDataDirectory("bugs"), bug.id + ".txt")
+            #created segmented bug type path
+            errDir = bug.error._cls
+            subFolderStr = "bugs/" + errDir
+            subFolder = os.path.join(config.getKwolaUserDataDirectory("bugs"), errDir) 
+            if not os.path.exists(subFolder):
+                getLogger().info(f"\n\n[{os.getpid()}] No folder for bug type : {subFolder}, creating...") 
+                try:
```

Comment:
I would have preferred if we had solved this entirely within config.getKwolaUserDataDirectory, rather then adding in this additional code here to create the folders if they don't exist. the ideal change here would be to modify getKwolaUserDataDirectory so that it can handle multiple levels of subfolders and create them all automatically